pkg/datapath/linux/config.go

```diff
@@ -0,0 +1,216 @@
+	fmt.Fprintf(fw, "#define METRICS_MAP_SIZE %d\n", metricsmap.MaxEntries)
+	fmt.Fprintf(fw, "#define POLICY_MAP_SIZE %d\n", policymap.MaxEntries)
+	fmt.Fprintf(fw, "#define IPCACHE_MAP %s\n", ipcachemap.Name)
+	fmt.Fprintf(fw, "#define IPCACHE_MAP_SIZE %d\n", ipcachemap.MaxEntries)
+	fmt.Fprintf(fw, "#define POLICY_PROG_MAP_SIZE %d\n", policymap.ProgArrayMaxEntries)
+	fmt.Fprintf(fw, "#define SOCKOPS_MAP_SIZE %d\n", sockmap.MaxEntries)
+
+	if option.Config.DatapathMode == option.DatapathModeIpvlan {
+		fmt.Fprintf(fw, "#define ENABLE_SECCTX_FROM_IPCACHE 1\n")
+	}
+
+	if option.Config.PreAllocateMaps {
+		fmt.Fprintf(fw, "#define PREALLOCATE_MAPS\n")
+	}
+
+	fmt.Fprintf(fw, "#define EVENTS_MAP %s\n", "cilium_events")
+	fmt.Fprintf(fw, "#define POLICY_CALL_MAP %s\n", policymap.CallMapName)
+	fmt.Fprintf(fw, "#define PROXY4_MAP cilium_proxy4\n")
+	fmt.Fprintf(fw, "#define PROXY6_MAP cilium_proxy6\n")
+	fmt.Fprintf(fw, "#define EP_POLICY_MAP %s\n", eppolicymap.MapName)
+	fmt.Fprintf(fw, "#define LB6_REVERSE_NAT_MAP cilium_lb6_reverse_nat\n")
+	fmt.Fprintf(fw, "#define LB6_SERVICES_MAP cilium_lb6_services\n")
+	fmt.Fprintf(fw, "#define LB6_RR_SEQ_MAP cilium_lb6_rr_seq\n")
+	fmt.Fprintf(fw, "#define LB4_REVERSE_NAT_MAP cilium_lb4_reverse_nat\n")
+	fmt.Fprintf(fw, "#define LB4_SERVICES_MAP cilium_lb4_services\n")
+	fmt.Fprintf(fw, "#define LB4_RR_SEQ_MAP cilium_lb4_rr_seq\n")
+
+	fmt.Fprintf(fw, "#define TRACE_PAYLOAD_LEN %dULL\n", option.Config.TracePayloadlen)
+	fmt.Fprintf(fw, "#define MTU %d\n", cfg.MtuConfig.GetDeviceMTU())
+
+	if option.Config.EnableIPv4 {
+		fmt.Fprintf(fw, "#define ENABLE_IPV4\n")
+	}
+
+	if option.Config.EnableIPv6 {
+		fmt.Fprintf(fw, "#define ENABLE_IPV6\n")
+	}
+
+	return fw.Flush()
+}
+
+func (l *linuxDatapath) writeNetdevConfig(w io.Writer, cfg datapath.DeviceConfiguration) {
+	fmt.Fprint(w, cfg.GetOptions().GetFmtList())
+	if option.Config.IsFlannelMasterDeviceSet() {
+		fmt.Fprint(w, "#define HOST_REDIRECT_TO_INGRESS\n")
+	}
+
+	// In case the Linux kernel doesn't support LPM map type, pass the set
+	// of prefix length for the datapath to lookup the map.
+	if ipcache.IPCache.MapType != bpf.BPF_MAP_TYPE_LPM_TRIE {
+		ipcachePrefixes6, ipcachePrefixes4 := cfg.GetCIDRPrefixLengths()
+
+		fmt.Fprint(w, "#define IPCACHE6_PREFIXES ")
+		for _, prefix := range ipcachePrefixes6 {
+			fmt.Fprintf(w, "%d,", prefix)
+		}
+		fmt.Fprint(w, "\n")
+		fmt.Fprint(w, "#define IPCACHE4_PREFIXES ")
+		for _, prefix := range ipcachePrefixes4 {
+			fmt.Fprintf(w, "%d,", prefix)
+		}
+		fmt.Fprint(w, "\n")
+	}
+}
+
+// WriteNetdevConfig writes the BPF configuration for the endpoint to a writer.
+func (l *linuxDatapath) WriteNetdevConfig(w io.Writer, cfg datapath.DeviceConfiguration) error {
+	fw := bufio.NewWriter(w)
+	l.writeNetdevConfig(fw, cfg)
+	return fw.Flush()
+}
+
+func mapPath(mapname string, e datapath.EndpointConfiguration) string {
+	return path.Base(bpf.MapPath(mapname + strconv.Itoa(int(e.GetID()))))
```

Comment:
This is the cause of the unit test failures in CI, as `bpf.MapPath()` may attempt to search for a BPF FS.